sw/device/tests/dif/dif_gpio_test.cc

```diff
@@ -0,0 +1,404 @@
+// Copyright lowRISC contributors.
```

Comment:
Is it better to add `sw/device/tests/mock` directory (for the sake of clarity that these are not meant to be run within a simulation platform) to put all mock tests together in one place?

Reply:
@gkelly @lenary what do you think of this, I understand that at the moment there wasn't concensus on where the unittests should reside?

Reply:
This debate came up before, and was covered in the DIF document:

> DIF_TEST_UNIT: Unit tests exist for the DIF in `sw/device/tests/dif` under `dif_<ip>_unittest.cc`

To summarise the points that lead us to that point:
- We do not want the exact implementation details of the tests reflected in the paths or filenames. These are unit tests (i.e. they only test DIF functions in isolation), but the fact they use mocks is an implementation detail.
- Opinions were expressed that there was little reason to split the DIF tests into further directories by type, and that it would be easier if all DIF tests were in the same directory.
- We do not intend for any C++ to run "on device", so we hoped that the filename extension would be another clue that this is a host-side test.

Does that help clarify the situation @sriyerg?

Reply:
Yup, fine with me.